File: Kısım2C/src/RastgeleKisi.c
#include "RastgeleKisi.h"

RastgeleKisi RastgeleKisiOlustur()
{
    RastgeleKisi this=(RastgeleKisi)malloc(sizeof(struct RASTGELEKISI));
    this->kisi=KisiOlustur();
    this->imeiKontrol=ImeiNoOlustur();
    this->tcKontrol=KimlikNoOlustur();
    this->islemYap=&_islemYap;
    this->RastgeleKisiYoket=&_RastgeleKisiYoket;
return this;
}

void _yeniKisiUret(const RastgeleKisi this)
{
    int kisiSayisi;
    printf("Kac Tane Kisi Uretmek Istiyorsunuz?");
    scanf("%d",&kisiSayisi);
        for(int i=0; i<kisiSayisi;i++)
            this->kisi->kisiUret(this->kisi);
}
void _dosyaKontrolEt(const RastgeleKisi this)
{
    this->imeiKontrol->imeiNoKontrol(this->imeiKontrol);
    this->tcKontrol->kimlikNoKontrolEt(this->tcKontrol);
}
void _islemYap(const RastgeleKisi this)   //public
{
    int secim;
    do{
       printf("1- Rastgele Kisi Uret\n");
        printf("2- Uretilmis Dosya Kontrol Et\n");
        printf("3- Kaydin tamamlanmasi ve cikis icin 3'e basin\n");
        scanf("%d",&secim);

       if(secim==1)
           _yeniKisiUret(this);
        if(secim==2)
            _dosyaKontrolEt(this);

    }while(secim!=3);
}
void _RastgeleKisiYoket(RastgeleKisi this)
{
    if(this==NULL)return;
    this->kisi->KisiYoket(this->kisi);
    this->imeiKontrol->ImeiNoYoket(this->imeiKontrol);
    this->tcKontrol->KimlikNoYoket(this->tcKontrol);
    free(this);
}

File: Kısım1Java/Kutuphane/RASTGELEKISIURET/src/MyPackage/RastgeleKisi.java
/**
*
* @author hakandurgay@gmail.com hakan durgay
* @since 26.04.2020
* <p>
* Açıklama
* </p>
*/
package MyPackage;

import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.Scanner;

public class RastgeleKisi {

    private int kisiSayisi;
   Kisi kisi;

    public RastgeleKisi() throws FileNotFoundException
    {
        kisi=new Kisi();
    }
    public void islemYap() throws IOException
    {
        int secim;
        do{
              System.err.println("1- Rastgele Kisi Uret");
              System.err.println("2- Uretilmis Dosya Kontrol Et");
              System.err.println("3- Cikis");
              Scanner menuSec=new Scanner(System.in);
              secim=menuSec.nextInt();
              if(secim==1)
                  kisiUret();
              else if(secim==2)
                  dosyaKontrolEt();
              
        }while(secim!=3);
    }
    private void kisiUret() throws IOException
    {
        
        System.out.println("Kac tane kisi uretmek istiyorsunuz");
        Scanner kisiAl=new Scanner(System.in);
        kisiSayisi=kisiAl.nextInt();
        
       
                for (int i = 0; i < kisiSayisi; i++) {       
                    kisi.kisiUret();     
           }
             kisi.kisileriDosyayaYaz();
    }
    private void dosyaKontrolEt() throws FileNotFoundException
    {
        KimlikNo tcKontrol=new KimlikNo();
        System.err.println("TC KIMLIK KONTROL");
        IMEINo imeiKontrol=new IMEINo();

        tcKontrol.tcKontrolEt();
                System.err.println("IMEI KONTROL");

        imeiKontrol.imeiNoKontrol();
    }
}
